sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/CosmosMultiHashTest.java

```diff
@@ -170,42 +195,120 @@ private void validateDocCRUDandQuery() throws Exception {
+            validateItemResponse(doc_current, response);
+        }
+
+        //Document Read - Negative tests
+        //Using incomplete partition key
+        PartitionKey partialPK = new PartitionKeyBuilder().add(doc.get("city").asText()).build();
+        try {
+            createdMultiHashContainer.readItem(doc.get("id").asText(), partialPK, ObjectNode.class);
+        } catch (CosmosException e) {
+            assertThat(e.getStatusCode()).isEqualTo(400);
+            assertThat(e.getMessage().contains("Partition key provided either doesn't correspond to definition in the collection or doesn't match partition key field values specified in the document.")).isTrue();
         }
 
-        // Query Tests.
+        //Read Many
+        List<Pair<String, PartitionKey>> pairList = new ArrayList<>();
+        List<CosmosItemIdentity> itemList = new ArrayList<>();
+        for (ObjectNode jsonNodes : docs) {
+            pairList.add(Pair.of(jsonNodes.get("id").asText(),
+                new PartitionKeyBuilder().add(jsonNodes.get("city").asText()).add(jsonNodes.get("zipcode").asText()).build()));
+            PartitionKey pkToUse = new PartitionKeyBuilder().add(jsonNodes.get("city").asText()).add(jsonNodes.get("zipcode").asText()).build();
+            itemList.add(new CosmosItemIdentity(pkToUse, jsonNodes.get("id").asText()));
+        }
+        FeedResponse<ObjectNode> documentFeedResponse = createdMultiHashContainer.readMany(itemList, ObjectNode.class);
+        assertThat(documentFeedResponse.getResults().size()).isEqualTo(pairList.size());
+        assertThat(documentFeedResponse.getResults().stream().map(jsonNode -> jsonNode.get("id").textValue()).collect(Collectors.toList()))
+            .containsAll(pairList.stream().map(p -> p.getLeft()).collect(Collectors.toList()));
+
+        //Read All
+        partitionKey = new PartitionKeyBuilder()
+            .add("Redmond")
+            .add("98053")
+            .build();
+        CosmosQueryRequestOptions cosmosQueryRequestOptions = new CosmosQueryRequestOptions();
+        cosmosQueryRequestOptions.setPartitionKey(partitionKey);
+
+        CosmosPagedIterable<ObjectNode> readAllResults =
+            createdMultiHashContainer.readAllItems(cosmosQueryRequestOptions, ObjectNode.class);
+        assertThat(readAllResults.stream().toArray().length).isEqualTo(2);
+
+        //Read All - Negative tests
+        //Using incomplete partition key
+        try {
+            readAllResults = createdMultiHashContainer.readAllItems(partialPK, ObjectNode.class);
+            readAllResults.stream().toArray();
+        } catch (CosmosException e) {
+            assertThat(e.getStatusCode()).isEqualTo(400);
+            assertThat(e.getMessage().contains("Partition key provided either doesn't correspond to definition in the collection or doesn't match partition key field values specified in the document.")).isTrue();
```

Comment:
why this one would fail? readAllItems should use query underlying, so would expect it to succeeded?

Reply:
oh nice good catch, going to need to re-verify this behavior now